File: 단계별로_풀어보기/4. 1차원 배열/9. 10811.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N, M = map(int, input().split())

resultArr = [i + 1 for i in range(N)]

for m in range(M):
  i, j = map(int, input().split())
  resultArr[i - 1: j] = resultArr[i - 1:j][::-1]

output(" ".join(str(r) for r in resultArr))

File: 단계별로_풀어보기/4. 1차원 배열/9. 10811.js
const fs = require('fs')
const inputs = fs.readFileSync('/dev/stdin').toString().split('\n')

const [N, M] = inputs[0].split(' ').map(Number)

const resultArr = Array(N)
  .fill(0)
  .map((_, idx) => idx + 1)

for (let idx = 0; idx < M; idx++) {
  const [i, j] = inputs[idx + 1].split(' ').map(Number)

  const partArr = resultArr.slice(i - 1, j)
  partArr.reverse()
  resultArr.splice(i - 1, j - i + 1, ...partArr)
}

process.stdout.write(`${resultArr.join(' ')}`)
